src/main/java/com/app/bookstore/model/Category.java

```diff
@@ -0,0 +1,28 @@
+package com.app.bookstore.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Data
+@Entity
+@Table(name = "categories")
+@SQLDelete(sql = "UPDATE categories SET is_deleted=true WHERE id=?")
+@Where(clause = "is_deleted=false")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
```

Comment:
Please replace it with @Column(nullable = false), they have one significant difference which is imprtant